judge/views/problem.py

```diff
@@ -201,6 +201,93 @@ def get_context_data(self, **kwargs):
                                           context['description'], 'problem')
         context['meta_description'] = self.object.summary or metadata[0]
         context['og_image'] = self.object.og_image or metadata[1]
+
+        context['can_vote'] = self.object.can_vote(user)
+        if context['can_vote']:
+            try:
+                context['vote'] = ProblemPointsVote.objects.get(voter=user.profile, problem=self.object)
+            except ObjectDoesNotExist:
+                context['vote'] = None
+        else:
+            context['vote'] = None
```

Comment:
I don't like this logic because it feels like it could be re-written and be cleaner, but I could just be wrong. Is there a `get_or_none` or something? Let's revisit this.

Reply:
It seems there's no `get_or_none`. Maybe there's no better way to write this?

Reply:
what is your opinion on `ProblemPointsVote.objects.filter(voter=user.profile, problem=self.object).first()`

Reply:
Let's not. As verbose as I feel this is, it's clear.